rpm/python3-oq-libs.spec.inc

```diff
@@ -90,27 +98,36 @@ Copyright (C) 2010-2018 GEM Foundation
 %setup -n %{oqformat} -T -D -a 1
 
 %install
-install -p -m 755 -d %{buildroot}/opt/openquake
-install -p -m 755 -d %{buildroot}/opt/openquake/bin
-helpers/whlsetup.sh -3 -b /opt/openquake/bin/python3.5 -d %{buildroot}/opt/openquake -s py -s py35 -s py35-extra
-install -p -m 755 -d %{buildroot}/opt/openquake/lib/python3.5/site-packages/openquake/libs
-install -p -m 644 openquake/libs/__init__.py %{buildroot}/opt/openquake/lib/python3.5/site-packages/openquake/libs
+install -p -m 755 -d %{buildroot}%{_prefix}
+install -p -m 755 -d %{buildroot}%{_bindir}
+helpers/whlsetup.sh -3 -b %{__python3} -d %{buildroot}%{_prefix} -s py -s py35 -s py35-extra
+install -p -m 755 -d %{buildroot}%{_prefix}/lib/python3.5/site-packages/openquake/libs
+install -p -m 644 openquake/libs/__init__.py %{buildroot}%{_prefix}/lib/python3.5/site-packages/openquake/libs
 # These aren't required anymore, but leaving them in case of PYTHONPATH is used
-install -p -m 644 openquake/__init__.py %{buildroot}/opt/openquake/lib/python3.5/site-packages/openquake
-install -p -m 644 openquake/__init__.py %{buildroot}/opt/openquake/lib64/python3.5/site-packages/mpl_toolkits
+install -p -m 644 openquake/__init__.py %{buildroot}%{_prefix}/lib/python3.5/site-packages/openquake
+install -p -m 644 openquake/__init__.py %{buildroot}%{_prefix}/lib64/python3.5/site-packages/mpl_toolkits
 
 %clean
 rm -rf %{buildroot}
 
+# Convert lib64 symlink into a real dir to avoid transaction conflicts during upgrades.
+# See: https://fedoraproject.org/wiki/Packaging:Directory_Replacement#Scriptlet_to_replace_a_symlink_to_a_directory_with_a_directory
+%pretrans -p <lua>
+path = "%{_libdir}"
+st = posix.stat(path)
+if st and st.type == "link" then
+  os.remove(path)
+end
+
```

Comment:
where is the magic here ? from 116 to 120 just check if path is a link and remove it, where is accomplished the rest of the conversion ? how is passed the target directory to ?

Reply:
The rest of magic is done by the RPM system

Reply:
>Replacing a symlink to a directory with a regular directory is much simpler, since there's no potential for accidentally removing files added externally. This scriptlet check the remove the symlink (there is no need to create the directory here, rpm will do so later when it is used)

From https://fedoraproject.org/wiki/Packaging:Directory_Replacement#Scriptlet_to_replace_a_symlink_to_a_directory_with_a_directory

Reply:
Ok, tnx.